contrib/drivers/ceph/replication.go

```diff
@@ -0,0 +1,407 @@
+		User: r.conf.Username,
+		Auth: []ssh.AuthMethod{
+			ssh.Password(r.conf.Password),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+	backupclient, error := ssh.Dial("tcp", r.conf.PeerdailIP, backupconfig)
+
+	if error != nil {
+		log.Error("Failed to dial: " + error.Error())
+
+	}
+
+	cephsession, err := cephclient.NewSession()
+
+	if err := cephsession.Run("systemctl stop ceph-rbd-mirror@ceph"); err != nil {
+		logs.Error(err)
+	}
+	cephsession.Close()
+
+	backupsession, err := backupclient.NewSession()
+
+	if err := backupsession.Run("systemctl stop ceph-rbd-mirror@remote"); err != nil {
+		logs.Error(err)
+	}
+
+	backupsession.Close()
+
+	return nil
+}
+
+// Start Replication
+func (r *ReplicationDriver) EnableReplication(opt *pb.EnableReplicationOpts) error {
+
+	cephconfig := &ssh.ClientConfig{
+		User: r.conf.Username,
+		Auth: []ssh.AuthMethod{
+			ssh.Password(r.conf.Password),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+	cephclient, err := ssh.Dial("tcp", r.conf.HostdailIP, cephconfig)
+	if err != nil {
+		log.Error("Failed to dial: " + err.Error())
+
+	}
+
+	cephsession, err := cephclient.NewSession()
+	if err != nil {
+		log.Error("Failed to create session: " + err.Error())
+
+	}
+
+	volumename := opensdsPrefix + opt.PrimaryVolumeId
+
+	cmd := "rbd mirror image enable rbd/" + volumename + " --pool rbd"
+
+	if err := cephsession.Run(cmd); err != nil {
+		log.Error("Failed to run: " + err.Error())
+
+	}
+
+	cephsession.Close()
+
+	return nil
+}
+
+// Stop Replication
+func (r *ReplicationDriver) DisableReplication(opt *pb.DisableReplicationOpts) error {
+
+	cephconfig := &ssh.ClientConfig{
+		User: r.conf.Username,
+		Auth: []ssh.AuthMethod{
+			ssh.Password(r.conf.Password),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+	cephclient, err := ssh.Dial("tcp", r.conf.HostdailIP, cephconfig)
+	if err != nil {
+		log.Error("Failed to dial: " + err.Error())
+
+	}
+	backupconfig := &ssh.ClientConfig{
+		User: r.conf.Username,
+		Auth: []ssh.AuthMethod{
+			ssh.Password(r.conf.Password),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+	backupclient, error := ssh.Dial("tcp", r.conf.PeerdailIP, backupconfig)
+
+	if error != nil {
+		log.Error("Failed to dial: " + error.Error())
+
+	}
+	volumename := opensdsPrefix + opt.PrimaryVolumeId
+
+	cephdemotesession, err := cephclient.NewSession()
+	if err != nil {
+	}
+
+	demotecmd := "rbd mirror image demote rbd/" + volumename
+	if err := cephdemotesession.Run(demotecmd); err != nil {
+		log.Error("Failed to run: " + err.Error())
+	}
+
+	cephdemotesession.Close()
+
+	time.Sleep(10 * time.Second)
+
+	backuppromotesession, err := backupclient.NewSession()
+	if err != nil {
+	}
+	promotecmd := "rbd mirror image promote rbd/" + volumename + " --cluster remote"
+	if err := backuppromotesession.Run(promotecmd); err != nil {
+		log.Error("Failed to run: " + err.Error())
+
+	}
+
+	backuppromotesession.Close()
+
+	time.Sleep(10 * time.Second)
```

Comment:
please do remove this . All the code should NOT have any arbitrary sleep(n). This is indicative of a bigger prob;em and creates many issues in the field when it is deployed. Use some polling mechanism to wait if needed.